apps/prairielearn/src/components/CommentPopover.html.ts

```diff
@@ -2,7 +2,11 @@ import { escapeHtml, html } from '@prairielearn/html';
 
 export function CommentPopover(comment: string | string[] | Record<string, any> | null) {
   let content = '';
-  if (comment === null) {
+  if (
+    !comment ||
+    (Array.isArray(comment) && comment.toString() === '[]') ||
```

Comment:
This is a pretty unconventional way to check for an empty array. `comment.length === 0` would be more appropriate.